tests/collator_test.py

```diff
@@ -29,13 +29,18 @@
 def test_get_collator(
     arch, has_features, has_target, expected_separate_features
 ):
+    dataset = datasets.BaseDataset()
     config = dataconfig.DataConfig(
         features_col=3 if has_features else 0,
         target_col=2 if has_target else 0,
     )
+    index = indexes.BaseIndex()
+    index.pad_idx = 1
```

Comment:
Index doesn't have a `pad_idx` or `source_vocab_size` in either your version or mine.  Not sure what's going on here.

Reply:
Okay I'll come back to this later.